test/boost/sstable_datafile_test.cc

```diff
@@ -3192,6 +3192,12 @@ SEASTAR_TEST_CASE(test_sstable_reclaim_memory_from_components) {
         BOOST_REQUIRE_EQUAL(sst_test.reclaim_memory_from_components(), total_reclaimable_memory);
         BOOST_REQUIRE_EQUAL(sst_test.total_reclaimable_memory_size(), 0);
         BOOST_REQUIRE_EQUAL(sst->filter_memory_size(), 0);
+
+        // Test sstable::reload_reclaimed_components() :
+        // Reloading should load the bloom filter back into memory
+        BOOST_REQUIRE_EQUAL(sst_test.reload_reclaimed_components(), total_reclaimable_memory);
+        BOOST_REQUIRE_EQUAL(sst_test.total_reclaimable_memory_size(), total_reclaimable_memory);
+        BOOST_REQUIRE_EQUAL(sst->filter_memory_size(), total_reclaimable_memory);
```

Comment:
This test is good to test the sstable API, but we need higher level tests, which add sstables to manager until we observe reclamation, then remove sstables, and observe previously evicted filters being reloaded.

Reply:
We should also attempt to test concurrent evict/reload. I know this is hard to do reliably, maybe we can use failure injection API, or fuzzing.

Reply:
Done.